models/user.js

```diff
@@ -0,0 +1,54 @@
+const mongoose = require('mongoose')
+const Schema = mongoose.Schema
+const { roles: { STUDENT, TEACHER, ADMIN } } = require('../consts/index');
+const bcrypt = require('bcrypt');
+const saltRounds = 10;
+
+const userSchema = new Schema(
+  {
+    role: {
+        type: String,
+        enum: [STUDENT, TEACHER, ADMIN],
+        required: true,
+        default: STUDENT
+    },
+    email: {
```

Comment:
i suggest that email should be unique + i'd like to add lowercase: true prop

Reply:
I see. But for which purposes we should use lowercase: true in email? Email is case sensitive and it can lead to some unexpected issues.